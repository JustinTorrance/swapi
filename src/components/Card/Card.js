import React, { Component } from 'react'
import './../../main.scss';

class Card extends Component {
  constructor({name, species, homeWorldPopulation, homeworld}) {
    super()
    this.state = {
      displayProfile: true
    }
  }

  handleCardClick = () => {
    this.setState({displayProfile: !this.state.displayProfile})
  }

  render() {
    if(this.state.displayProfile) {
      return (
        <div className='Card' onClick={() => this.handleCardClick()}>
          <div className='card-profile'>
            <button className='card-favorite'><i className="fas fa-heart"></i></button>
            <img className='card-image' src='https://moviewriternyu.files.wordpress.com/2015/07/chewy-2.png' alt='wookie' />
            <h3>{this.props.name}</h3>
          </div>
        </div>
      )      
    } else {
      return (
        <div className='Card' onClick={() => this.handleCardClick()}>
          <div className='card-info'>
            <table className='card-info-table'>
              <tbody>
                <tr><th className='card-info-table-name'>{this.props.name}</th></tr>
                <tr><td className='card-info-table-species'>Species: {this.props.species}</td></tr>
                <tr><td className='card-info-table-language'>Homeworld: {this.props.homeworld}</td></tr>
                <tr><td className='card-info-table-data'>Homeworld Population: {this.props.homeWorldPopulation}</td></tr>
              </tbody>
            </table>
          </div>
        </div>
      )      
    } 
  }

}

export default Card;

