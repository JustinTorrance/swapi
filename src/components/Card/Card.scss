.Card {
  box-sizing: border-box;
  border: 4px solid #fff;
  border-radius: 20px;
  width: 100%;
}

.card-favorite {
  float: right;
  margin: 10px;
  margin-left: 100%;
}

.card-image {
  border: 1px solid #fff;
  border-radius: 10px;
  margin: 10px 5px 0 5px;
  padding: 20px 0 20px 0;
  width: 80%;
}

.card-profile-name {
  font-family: 'Jedi-Rounded', sans-serif;
  font-size: 1.5rem;
  letter-spacing: 0.1rem;
  margin: 50px 15px 100px 15px;
}

.card-info {
  height: 100%;
}

.card-info-table {
  height: 100%;
  margin: 0 auto;
  text-align: left;
  width: 80%;
}

.card-info-table-row {
  border-bottom: 1px solid #fff;
}

.card-info-table-row-bottom {
  border-bottom: none;
}
