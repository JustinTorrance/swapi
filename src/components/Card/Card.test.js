import React from 'react'; 
import { shallow, mount } from 'enzyme';
import Card from './Card.js';

describe('Card', () => {
  it('should match the snapshot', () => {
    const mockCardData = {
      Name:'Luke',
      Species:'Human',
      Homeworld:'Tattooine',
      HomeworldPopulation: 150000
    }

    const wrapper = shallow(
      <Card 
        addFavorite={jest.fn()}
        removeFavorite={jest.fn()}
        cardCategory='people'
        card={mockCardData}
      />
    ); 
    expect(wrapper).toMatchSnapshot();
  });

});