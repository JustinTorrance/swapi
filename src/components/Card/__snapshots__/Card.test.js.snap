// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[MockFunction]}
    card={
      Object {
        "Homeworld": "Tattooine",
        "HomeworldPopulation": 150000,
        "Name": "Luke",
        "Species": "Human",
      }
    }
    cardCategory="people"
    removeFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="card-favorite"
          onClick={[Function]}
        >
          <i
            className="card-favorite-icon fas fa-heart"
            onClick={[Function]}
          />
        </button>,
        <div
          className="card-profile"
          onClick={[Function]}
        >
          <img
            alt="wookie"
            className="card-image"
            src="https://moviewriternyu.files.wordpress.com/2015/07/chewy-2.png"
          />
          <h3
            className="card-profile-name"
          >
            luke
          </h3>
        </div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="card-favorite-icon fas fa-heart"
            onClick={[Function]}
          />,
          "className": "card-favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "card-favorite-icon fas fa-heart",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="wookie"
              className="card-image"
              src="https://moviewriternyu.files.wordpress.com/2015/07/chewy-2.png"
            />,
            <h3
              className="card-profile-name"
            >
              luke
            </h3>,
          ],
          "className": "card-profile",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "wookie",
              "className": "card-image",
              "src": "https://moviewriternyu.files.wordpress.com/2015/07/chewy-2.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "luke",
              "className": "card-profile-name",
            },
            "ref": null,
            "rendered": "luke",
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="card-favorite"
            onClick={[Function]}
          >
            <i
              className="card-favorite-icon fas fa-heart"
              onClick={[Function]}
            />
          </button>,
          <div
            className="card-profile"
            onClick={[Function]}
          >
            <img
              alt="wookie"
              className="card-image"
              src="https://moviewriternyu.files.wordpress.com/2015/07/chewy-2.png"
            />
            <h3
              className="card-profile-name"
            >
              luke
            </h3>
          </div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="card-favorite-icon fas fa-heart"
              onClick={[Function]}
            />,
            "className": "card-favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "card-favorite-icon fas fa-heart",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="wookie"
                className="card-image"
                src="https://moviewriternyu.files.wordpress.com/2015/07/chewy-2.png"
              />,
              <h3
                className="card-profile-name"
              >
                luke
              </h3>,
            ],
            "className": "card-profile",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "wookie",
                "className": "card-image",
                "src": "https://moviewriternyu.files.wordpress.com/2015/07/chewy-2.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "luke",
                "className": "card-profile-name",
              },
              "ref": null,
              "rendered": "luke",
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
